### PR TITLE
feat(specs): Dencun P2P & Engine API changes

### PR DESCRIPTION
**Pull request by danyalprout**
**Description**

Specify the engine API and p2p changes for Dencun.

### CODE REVIEW

**Comment by danyalprout** [comment]
Would love to get peoples thoughts on what value the parent beacon block root should be when the L1 does not support it.

e.g. if Ecotone activates before Cancun, or someone is using an Alt-DA layer.

File: specs/derivation.md [hunk 1]
```diff
@@ -798,24 +798,41 @@ To interact with the engine, the [execution engine API][exec-engine] is used, wi
...
 [`engine_newPayloadV2`]: exec-engine.md#engine_newpayloadv2
+[`engine_newPayloadV3`]: exec-engine.md#engine_newpayloadv3
+
+The execution payload is an object of type [`ExecutionPayloadV3`][eth-payload].
 
-The execution payload is an object of type [`ExecutionPayloadV2`][eth-payload].
+[eth-payload]: https://github.com/ethereum/execution-apis/blob/main/src/engine/cancun.md
 
-[eth-payload]: https://github.com/ethereum/execution-apis/blob/main/src/engine/shanghai.md#payloadattributesv2
+The `ExecutionPayload` has the following requirements:
```
Comment (tynes) [inline specs/derivation.md:824]: I am a bit confused here what the difference between post canyon and pre ecotone means.

Perhaps we can do
- Bedrock
- Canyon
- Ecotone

File: specs/derivation.md [hunk 2]
```diff
@@ -907,8 +924,8 @@ To process unsafe payloads, the payload must:
 
 The payload is then processed with a sequence of:
 
-- `engine_newPayloadV2`: process the payload. It does not become canonical yet.
-- `engine_forkchoiceUpdatedV2`: make the payload the canonical unsafe L2 head, and keep the safe/finalized L2 heads.
+- `engine_newPayloadV3`: process the payload. It does not become canonical yet. (pre-Ecotone payloads MUST use V2)
```
Comment (tynes) [inline specs/derivation.md]: Instead of pre ecotone, perhaps use Bedrock and Canyon

File: specs/exec-engine.md [hunk 3]
```diff
@@ -15,10 +15,12 @@
     - [Pre-Ecotone](#pre-ecotone)
     - [Ecotone L1-Cost fee changes (EIP-4844 DA)](#ecotone-l1-cost-fee-changes-eip-4844-da)
 - [Engine API](#engine-api)
-  - [`engine_forkchoiceUpdatedV2`](#engine_forkchoiceupdatedv2)
-    - [Extended PayloadAttributesV2](#extended-payloadattributesv2)
+  - [`engine_forkchoiceUpdatedV3`](#engine_forkchoiceupdatedv3)
```
Comment (tynes) [inline specs/exec-engine.md:18]: If v2 is still required for bedrock and canyon then should we add a new section and leave the old section instead?

File: specs/exec-engine.md [hunk 4]
```diff
@@ -196,18 +198,19 @@ Within the rollup, the types of forkchoice updates translate as:
...
 
-#### Extended PayloadAttributesV2
+#### Extended PayloadAttributesV3
 
-[`PayloadAttributesV2`][PayloadAttributesV2] is extended to:
+[`PayloadAttributesV3`][PayloadAttributesV3] is extended to:
 
 ```js
-PayloadAttributesV2: {
+PayloadAttributesV3: {
```
Comment (tynes) [inline specs/exec-engine.md:208]: v2 is still required for bedrock and canyon?

File: specs/exec-engine.md (hunk 3 above)
Comment (danyalprout) [reply specs/exec-engine.md:18]: I believe we only need v2 for `newPayload`. `forkchoiceUpdated` and `getPayload` the V3 API's are backwards compatible and we can use them for all blocks.

File: specs/exec-engine.md (hunk 4 above)
Comment (danyalprout) [reply specs/exec-engine.md:208]: You can use the V3 payload, you just have to omit the parentBeaconBlockRoot. Similar to when we went from Bedrock to Canyon, you could use V2 for Bedrock, but had to make sure withdrawals == nil.

**Pull request closed without merging**